proxy/src/main/java/com/wavefront/agent/PushAgent.java

```diff
@@ -497,6 +507,30 @@ protected void startTraceListener(final String strPort, ReportableEntityHandlerF
         .withChildChannelOptions(childChannelOptions), "listener-plaintext-trace-" + port);
   }
 
+  protected void startTraceJaegerListener(String strPort, ReportableEntityHandlerFactory handlerFactory) {
+    startAsManagedThread(() -> {
+      activeListeners.inc();
+      try {
+        TChannel server = new TChannel.Builder("jaeger-collector").
+            setServerHost(InetAddress.getLoopbackAddress()).
+            setServerPort(Integer.valueOf(strPort)).
+            build();
+        server.
+            makeSubChannel("jaeger-collector", Connection.Direction.IN).
+            register("Collector::submitBatches", new JaegerThriftCollectorHandler(strPort, handlerFactory,
+                traceDisabled));
+        server.listen().channel().closeFuture().sync();
+        server.shutdown(false);
+      } catch (InterruptedException e) {
+        logger.info("Listener on port " + strPort + " shut down.");
+      } catch (Exception e) {
+        logger.log(Level.SEVERE, "Jaeger trace collector exception", e);
+      } finally {
+        activeListeners.dec();
```

Comment:
@basilisk487 shouldn't activeListeners be a gauge? not part of this PR but you can think about changing this to a gauge in subsequent PR.